tests/BuildScriptGenerator.Tests/DotnetCore/GlobalJsonSdkResolverTest.cs

```diff
@@ -0,0 +1,1217 @@
+                RollForward = GlobalJsonSdkResolver.Major,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Equal(expectedVersion, actual);
+        }
+
+        [Fact]
+        public void Major_ReturnsNextMajorAndNextMinorAndNextHigherFeatureAndLatestPatchLevel_IfSpecifiedMajorIsNotPresent()
+        {
+            // Arrange
+            var expectedVersion = "4.1.101";
+            var availableVersions = new[]
+            {
+                "2.1.100",
+                "4.1.100",
+                "4.1.101",
+                "4.2.200",
+                "4.2.201",
+            };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "3.1.100",
+                RollForward = GlobalJsonSdkResolver.Major,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Equal(expectedVersion, actual);
+        }
+
+        [Fact]
+        public void Major_ReturnsNull_IfNextMajorAndNextMinorAndNextHigherFeatureAndLatestPatchLevel_IsNotPresent()
+        {
+            // Arrange
+            var availableVersions = new[]
+            {
+                "2.1.100",
+                "4.1.100",
+                "4.1.101",
+                "4.2.200",
+                "4.2.201",
+            };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "4.3.100",
+                RollForward = GlobalJsonSdkResolver.Major,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Null(actual);
+        }
+
+        [Fact]
+        public void LatestPatch_ReturnsSameVersion_IfNoLatestPatchVersionIsAvailable()
+        {
+            // Arrange
+            var expectedVersion = "3.1.100";
+            var availableVersions = new[] { "2.1.100", "3.1.100" };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "3.1.100",
+                RollForward = GlobalJsonSdkResolver.LatestPatch,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Equal(expectedVersion, actual);
+        }
+
+        [Fact]
+        public void LatestPatch_OnlyConsidersPatchWithinSameFeature()
```

Comment:
Not sure about this one: `LatestPatch_ReturnsSamePatchVersion_IfPatchVersionsWithinSameFeatureAreNotAvailable`?

Reply:
Nice!